.gitignore

```diff
@@ -1,3 +1,4 @@
+/composer.lock
```

Comment:
I understand what your saying, but both come with advantages and disadvantages. I'm happy to in fact have both. A separate build job that removes the lock file and tests the code against whatever the latest packages are would be OK?